lib/ansible/modules/cloud/openstack/os_port.py

```diff
@@ -228,21 +239,35 @@ def _needs_update(module, port, cloud):
                       'device_id',
                       'binding:vnic_type',
                       'port_security_enabled']
-    compare_dict = ['allowed_address_pairs',
-                    'extra_dhcp_opts']
+    compare_list_dict = ['allowed_address_pairs',
+                         'extra_dhcp_opts']
     compare_list = ['security_groups']
 
     for key in compare_simple:
         if module.params[key] is not None and module.params[key] != port[key]:
             return True
-    for key in compare_dict:
-        if module.params[key] is not None and module.params[key] != port[key]:
-            return True
     for key in compare_list:
         if module.params[key] is not None and (set(module.params[key]) !=
                                                set(port[key])):
             return True
 
+    for key in compare_list_dict:
+        if module.params[key] is not None:
+            if not port[key]:
+                return True
+
+            # sort dicts in list
+            port_ordered = [OrderedDict(sorted(d.items())) for d in port[key]]
+            param_ordered = [OrderedDict(sorted(d.items())) for d in module.params[key]]
+
+            for d in param_ordered:
```

Comment:
these would loop over the list and return `True` (breaking out of the function) on the first port not in `port_ordered`. Is that what we're going for?

Reply:
Yes, this will trigger an update if there's any differences between what's set in the parameters and what's returned by openstack